Add new and update existing popover api docs (#26413)
mdn/content: files/en-us/web/api/toggleevent/index.md

> {{APIRef("UI Events")}}

Reviewer: So the `toggle` event is "coalesced" meaning if multiple of them are fired before the event loop has a chance to cycle, only a single `toggle` event will be fired. (Spec is [here](https://html.spec.whatwg.org/multipage/popover.html#queue-a-popover-toggle-event-task).) And in that case, the interpretation of the `oldState` and `newState` parameters is... interesting.

I say this for your information. It might be too much information to include on MDN. Then again, a developer encountering this behavior might wish it was here.

The example is:

```javascript
p.addEventListener('toggle',()=>{});
p.showPopover();
p.hidePopover();
// `toggle` only fires once
```

(I didn't verify the above example, but it should be true.)

Author: I think this is worth including, but I've included it on the pages for the HTMLElement beforetoggle/toggle events, as I think it is more useful there. Added.

{{APIRef("UI Events")}}{{SeeCompatTable}}

The **`ToggleEvent`** interface represents an event notifying the user when a popover element's state toggles between showing and hidden.

> It is the event object for the {{domxref("HTMLElement.beforetoggle_event", "beforetoggle")}} and {{domxref("HTMLElement.toggle_event", "toggle")}} events, which fire on popovers when their state changes.

Reviewer 2: But (I think) only for those events when the target is a popover. So maybe something like:

```suggestion
It is the event object for the {{domxref("HTMLElement.beforetoggle_event", "beforetoggle")}} and {{domxref("HTMLElement.toggle_event", "toggle")}} events when the event target is a popover.
```

Author: Yes, but the `<details>` version seems to be a different, unrelated implementation. I've updated the text slightly differently, adding a note to explain that `ToggleEvent` is not related to that version of `toggle`, and how they differ.

Reviewer: See my comment above.
